### PR TITLE
provide PPL for querying system indices

### PR DESCRIPTION
**Pull request by xinyual**
### Description
provide PPL for querying system indices
 
### Issues Resolved
provide PPL for querying system indices. Will address recursively replacement.
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/agent/tools/PPLTool.java [hunk 1]
```diff
@@ -63,6 +65,9 @@ public class PPLTool implements Tool {
 
     public static final String TYPE = "PPLTool";
 
+    public static final List<String> SYSTEM_INDICES = List
+        .of(ML_MODEL_GROUP_INDEX, ML_MODEL_INDEX, ML_TASK_INDEX, ML_CONNECTOR_INDEX, ML_CONFIG_INDEX, ML_MODEL_CONTROLLER_INDEX);
```
Comment (zane-neo) [inline src/main/java/org/opensearch/agent/tools/PPLTool.java:69]: There could be more indices like: agent, memory etc, using regex like `.plugins-ml-*` is more flexible to adapt to future changes. BTW, do we need to consider other system indices which are not ml-common's?

File: src/main/java/org/opensearch/agent/tools/PPLTool.java [hunk 2]
```diff
@@ -97,6 +102,11 @@ public <T> void run(Map<String, String> parameters, ActionListener<T> listener)
         if (StringUtils.isBlank(indexName) || StringUtils.isBlank(question)) {
             throw new IllegalArgumentException("Parameter index and question can not be null or empty.");
         }
+        for (String systemIndex : SYSTEM_INDICES) {
+            if (indexName.contains(systemIndex)) {
+                throw new IllegalArgumentException("We cannot search system indices " + indexName);
```
Comment (zane-neo) [inline src/main/java/org/opensearch/agent/tools/PPLTool.java:107]:
```suggestion
                throw new IllegalArgumentException("PPLTool doesn't support searching system indices, current searching index name: " + indexName);
```

File: src/main/java/org/opensearch/agent/tools/PPLTool.java (hunk 1 above)
Comment (xinyual) [reply src/main/java/org/opensearch/agent/tools/PPLTool.java:69]: I think so. Is there a pattern to match all system indices?

Comment (zane-neo) [reply src/main/java/org/opensearch/agent/tools/PPLTool.java:69]: Looks like all system indices are starting with `.`, here are some examples: `.opendistro-alerting-config`,`.opensearch-observability`, `.replication-metadata-store`. Check if starting with `.` could be safe and also fine-tune the error message.

Comment (xinyual) [reply src/main/java/org/opensearch/agent/tools/PPLTool.java]: Already done.

**Review by zane-neo — APPROVED** [review]

**Pull request merged**